bitrepository-core/src/main/java/org/bitrepository/common/utils/ChecksumUtils.java

```diff
@@ -267,13 +267,13 @@ public static boolean requiresSalt(ChecksumType algorithm) throws NoSuchAlgorith
      */
     public static ChecksumSpecTYPE getDefault(Settings settings) {
         ChecksumSpecTYPE res = new ChecksumSpecTYPE();
-        res.setChecksumType(ChecksumType.valueOf(
-                settings.getRepositorySettings().getProtocolSettings().getDefaultChecksumType()));
+        res.setChecksumType(ChecksumType.valueOf(settings.getRepositorySettings().getProtocolSettings().getDefaultChecksumType()));
 
         try {
             verifyAlgorithm(res);
         } catch (NoSuchAlgorithmException e) {
-            throw new IllegalArgumentException("The settings contains an invalid default checksum specification.", e);
+            System.err.println("The default settings file contains an invalid checksum spec.");
+            System.exit(1);
```

Comment:
I much prefer throwing the exception to this. I think System.exit(1) at a higher level e.g. in CommandLineClient is fine, but getDefault() is used a lot throughout the code where it would be nicer to have a stacktrace from an exception.

Besides, I don't even know if logging picks up System.err (it can probably be configured to though).

Reply:
Also, I feel the error message was more accurate the way it was before :^)

Reply:
I can't remember why I changed this but there was a good reason related to the IllegalArgumentException and how it was propagated, but I think I might have changed that logic somewhere and forgot to change this back